src/utils/exponentialBackoff.ts

```diff
@@ -0,0 +1,12 @@
+async function exponentialBackoff<T>(fn: () => Promise<T>, retries = 5, delay = 1000): Promise<T> {
```

Comment:
2^5=32 seconds is too long. The http request will likely timeout. 2^3=8 seconds is more than enough I think.
```suggestion
async function exponentialBackoff<T>(fn: () => Promise<T>, retries = 3, delay = 1000): Promise<T> {
```